particles.html

```diff
@@ -316,15 +330,15 @@
         velocity += acceleration;
         velocity *= 0.9999;
 
-        positionsOut[index].xyz = position + velocity;
-        velocityOut[index].xyz = velocity;
+        positionsOut[index] = vec4(position + velocity, 1);
+        velocityOut[index] = vec4(velocity, 1);

```

Comment:
I'm not totally sure what's going on here. With the vec3 write I get:
```
1 error(s) generated while compiling the shader:
error: cannot initialize let of type 'ptr<storage, vec4<f32>>' with value of type 'ptr<storage, vec4<f32>, read_write>'
```

Curious if you know what that's about!

Reply:
Hmm... that's weird. I wonder if it's an implementation bug.

Reply:
I haven't dug into it yet, but this issue looks like it's related to WGSL constraints/semantics: https://www.w3.org/TR/WGSL/#memory-view-types